frontend/src/stateless/index.ts

```diff
@@ -252,7 +252,45 @@ export function findKubeconfigByClusterName(clusterName: string): Promise<string
             );
 
             if (matchingKubeconfig || matchingContext) {
-              resolve(kubeconfig);
+              const contextToUpdate = matchingKubeconfig || matchingContext;
+              const extensions = contextToUpdate?.context.extensions || [];
+              let headlampExtension = extensions.find(
+                extension => extension.name === 'headlamp_info'
```

Comment:
Why do stateless clusters need the `headlamp_info` extension? This extension's purpose was to avoid modifying the cluster name of clusters coming from "external" (to headlamp) kubeconfigs, and we were even talking the other day about whether we should even do this at all for those. But for anything that "belongs" or is defined in Headlamp, like stateless clusters, we should just change the actual context name when the user chooses to update it.